### DISCUSSION TITLE
Purge Schema Cache?

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#12357] [Archive]
HI,
I've run into this (and other similar issues), after updating the schema manually through SQL.
Now it appears that some actions are broken: this screenshot comes after uploading a file:

![image](https://user-images.githubusercontent.com/56647386/159938362-860da13f-3534-42eb-af57-6f55dde4eb46.png)

I had added that field into the file schema initially as I needed it for some type of relationship setup that I was trying to implement.
Once I tried to remove that field from the back-end interface I kept on being prompted with an error about
"not being able to delete the field because they KEY was used somewhere else..." I don't remember exactly: eventually I ended up deleting the field from the DB directly.

So now the schema has obviously a problem. Can I fix that with a schema CACHE purge?
and if so, how can I do that...

So if I try to re-add the field above to the system collection "files",
this is what I get:
![image](https://user-images.githubusercontent.com/56647386/159950903-06e81f90-43f7-4d4e-8d79-60289b49531c.png)

---

Update:
I was able to purge the schema cache using postman and a static admin token. 
But problem persists..

**Comment by rijkvanzanten** [comment]
Clearing the caches can be done through the utils endpoint: https://docs.directus.io/reference/system/utilities/#clear-the-internal-cache Alternatively, restarting Directus will also clear the caches on first startup.

If you're planning to make updates directly in the database, you could consider disabling the schema cache to make sure everything stays up to date by setting `CACHE_SCHEMA=false` (https://docs.directus.io/configuration/config-options/#cache)
